src/constants/passwordPolicy.js

```diff
@@ -0,0 +1,30 @@
+const passwordPolicy = {
+  minLength: 10,
+  hasUpperCase: true,
+  hasLowerCase: true,
+  hasNumber: true,
+  hasSpecialChar: true,
+};
+
+const validatePassword = (password) => {
+  const { minLength, hasUpperCase, hasLowerCase, hasNumber, hasSpecialChar } = passwordPolicy;
+
+  if (password.length < minLength) {
+    return `Password must be at least ${minLength} characters long.`;
+  }
+  if (hasUpperCase && !/[A-Z]/.test(password)) {
+    return "Password must contain at least one uppercase letter.";
+  }
+  if (hasLowerCase && !/[a-z]/.test(password)) {
+    return "Password must contain at least one lowercase letter.";
+  }
+  if (hasNumber && !/[0-9]/.test(password)) {
+    return "Password must contain at least one number.";
+  }
+  if (hasSpecialChar && !/[!@#$%^&*(),.?":{}|<>]/.test(password)) {
+    return "Password must contain at least one special character.";
+  }
+  return null;
+};
```

Comment:
If the password fails multiple validation rules, the current implementation only returns the first validation error it encounters. This means that the user would receive feedback about one specific issue at a time, which can be less efficient but I thought it might be less confusing for the user although he might have to go through multiple rounds of corrections.
To improve the user experience, we can modify the validatePassword method to collect all validation errors and return them together. Then, the user can see all the issues with their password at once and address them in a single attempt. Is it better?

Reply:
That would be better.

I've seen implementations that list the validation rules and display check-marks or X's to indicate compliance as the user types the password.